Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm

```diff
@@ -368,7 +368,26 @@
     [screenTimeWebpageController setURLIsPlayingVideo:value];
 }
 
+void PageClientImplCocoa::showScreenTimeShieldForWindow()
+{
+    [m_webView _showScreenTimeShieldForWindow];
+}
+
+#endif
+
+void PageClientImplCocoa::viewIsBecomingVisible()
```

Comment:
`class PageClientImpl final : public PageClientImplCocoa` overrides this again on iOS, so this doesn't seem quite right, as none of this code would run on iOS?

You need to call `PageClientImplCocoa::viewIsBecomingVisible()` in `PageClientImpl::viewIsBecomingVisible()`.

Reply:
Or alternatively, delete that override and move the logic into here:

```
#if ENABLE(PAGE_LOAD_OBSERVER)
    if (RetainPtr webView = this->webView())
        [webView _updatePageLoadObserverState];
#endif
```